cmd/jaeger/internal/extension/remotesampling/config.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package remotesampling
+
+import (
+	"errors"
+	"reflect"
+	"time"
+
+	"go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/confighttp"
+)
+
+var (
+	errNoSource       = errors.New("no sampling strategy specified, has to be either 'adaptive' or 'file'")
+	errMultipleSource = errors.New("only one sampling strategy can be specified, has to be either 'adaptive' or 'file'")
+)
+
+type FileConfig struct {
+	// File specifies a local file as the strategies source
+	Path string `mapstructure:"path"`
+}
+
+type AdaptiveConfig struct {
+	// name of the strategy storage defined in the jaegerstorage extension
+	StrategyStore string `mapstructure:"strategy_store"`
+
+	// InitialSamplingProbability is the initial sampling probability for all new operations.
+	InitialSamplingProbability float64 `mapstructure:"initial_sampling_probability"`
+
+	// AggregationBuckets is the total number of aggregated throughput buckets kept in memory, ie. if
+	// the CalculationInterval is 1 minute (each bucket contains 1 minute of thoughput data) and the
+	// AggregationBuckets is 3, the adaptive sampling processor will keep at most 3 buckets in memory for
+	// all operations.
+	// TODO(wjang): Expand on why this is needed when BucketsForCalculation seems to suffice.
+	AggregationBuckets int `mapstructure:"aggregation_buckets"`
+
+	// MinSamplesPerSecond determines the min number of traces that are sampled per second.
+	// For example, if the value is 0.01666666666 (one every minute), then the sampling processor will do
+	// its best to sample at least one trace a minute for an operation. This is useful for low QPS operations
+	// that may never be sampled by the probabilistic sampler.
+	MinSamplesPerSecond float64 `mapstructure:"min_samples_per_second"`
+
+	// LeaderLeaseRefreshInterval is the duration to sleep if this processor is elected leader before
+	// attempting to renew the lease on the leader lock. NB. This should be less than FollowerLeaseRefreshInterval
+	// to reduce lock thrashing.
+	LeaderLeaseRefreshInterval time.Duration `mapstructure:"leader_lease_refresh_interval"`
+
+	// FollowerLeaseRefreshInterval is the duration to sleep if this processor is a follower
+	// (ie. failed to gain the leader lock).
+	FollowerLeaseRefreshInterval time.Duration `mapstructure:"follower_lease_refresh_interval"`
+}
+
+type Config struct {
+	File     FileConfig     `mapstructure:"file"`
+	Adaptive AdaptiveConfig `mapstructure:"adaptive"`
+	HTTP     HTTPConfig     `mapstructure:"http"`
+	GRPC     GRPCConfig     `mapstructure:"grpc"`
+}
```

Comment:
fyi I opened a ticket https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33232

Reply:
Yes, let's do the same. We should define all 4 fields in the config as pointers, so that both File/Adaptive can be optional, and HTTP/GRPC can also be disabled independently.

We need to have a unit test for this behavior in the config, i.e. feed it different YAML strings and check that the corresponding sections are there or not there. I think we also have other configs that could use this pattern, I will take a look.

Lastly, the Validate function below should change to reflect this pointer change, the logic there can be different, e.g. instead of `if cfg.File.Path != "" && cfg.Adaptive.StrategyStore != ""` we should check if the whole struct is defined or not.

I also think the sub-structs should have their own Validate method that can be called from the main Validate if the struct is not nil. For example, if someone defines `file:` without any arguments we should raise an error that the config file name is missing (I think we can use struct tags to add these requirements).